tests/std/tests/P0355R7_calendars_and_time_zones_dates/test.cpp

```diff
@@ -0,0 +1,930 @@
+    assert(2020y / April > 2019y / April);
+    assert(2020y / April > 2020y / March);
+
+    assert(ym + months{2} == 2020y / March);
+    assert(months{2} + ym == 2020y / March);
+
+    assert(ym - months{2} == 2019y / November);
+    assert(ym - 2019y / January == months{12});
+
+    assert(ym + years{2} == 2022y / January);
+    assert(years{2} + ym == 2022y / January);
+
+    assert(ym - years{2} == 2018y / January);
+
+    if (is_constant_evaluated()) {
+        static_assert((2020y / 1).ok());
+        static_assert(!(2020y / 13).ok());
+        static_assert(!(32768y / 1).ok());
+    } else {
+        for (int y = y_min - 1; y <= y_max + 1; ++y) {
+            for (auto m = 0u; m <= 255u; ++m) {
+                const auto ym2 = year{y} / month{m};
+                if (y == y_min - 1 || y == y_max + 1) {
+                    assert(!ym2.ok());
+                } else if (m >= 1 && m <= 12) {
+                    assert(ym2.ok());
+                } else {
+                    assert(!ym2.ok());
+                }
+            }
+        }
+    }
+}
+
+constexpr void year_month_day_test() {
+    year_month_day ymd1{2020y / January / 1d};
+
+    static_assert(noexcept(year_month_day{}));
+    static_assert(noexcept(year_month_day{year{}, month{}, day{}}));
+    static_assert(noexcept(year_month_day{year_month_day_last{year{}, month_day_last{January}}}));
+    static_assert(noexcept(year_month_day{sys_days{}}));
+    static_assert(noexcept(year_month_day{local_days{}}));
+
+    static_assert(noexcept(ymd1 += months{}));
+    static_assert(noexcept(ymd1 -= months{}));
+    static_assert(noexcept(ymd1 += years{}));
+    static_assert(noexcept(ymd1 -= years{}));
+
+    static_assert(noexcept(ymd1.year()));
+    static_assert(noexcept(ymd1.month()));
+    static_assert(noexcept(ymd1.day()));
+
+    static_assert(noexcept(static_cast<sys_days>(ymd1)));
+    static_assert(noexcept(static_cast<local_days>(ymd1)));
+    static_assert(noexcept(ymd1.ok()));
+
+    static_assert(noexcept(ymd1 == ymd1));
+    static_assert(noexcept(ymd1 <=> ymd1));
+
+    static_assert(noexcept(ymd1 + months{}));
+    static_assert(noexcept(months{} + ymd1));
+    static_assert(noexcept(ymd1 - months{}));
+
+    static_assert(noexcept(ymd1 + years{}));
+    static_assert(noexcept(years{} + ymd1));
+    static_assert(noexcept(ymd1 - years{}));
+
+    assert(ymd1.year() == 2020y);
+    assert(ymd1.month() == January);
+    assert(ymd1.day() == 1d);
+
+    year_month_day ymd2{2020y / January / last};
+    assert(ymd2.year() == 2020y);
+    assert(ymd2.month() == January);
+    assert(ymd2.day() == 31d);
+
+    year_month_day epoch{sys_days{}};
+    assert(epoch == year_month_day{sys_days{epoch}});
+    assert(epoch.year() == 1970y);
+    assert(epoch.month() == January);
+    assert(epoch.day() == 1d);
+
+    local_days ldp;
+    sys_days sys{ldp.time_since_epoch()};
+    year_month_day ymld{ldp};
+    assert((ymld == year_month_day{sys}));

```

Comment:
There are a few `assert` lines with extra parentheses. The problematic scenario is when we have code with angle brackets (like `pair<int, int>`) or braces (like `pair{11, 22}`) because the preprocessor doesn't think that `<>` and `{}` are special, so it consumes the `,`. (The preprocessor does know how to match `()` though.)